### PR TITLE
Update scrimage to 4.x from 2.x & fix copying resources

### PR DESCRIPTION
**Pull request by sloshy**
Scrimage, a dependency of sbt-microsites, has actually been significantly out of date for some time, probably due to a dependency name change from `scrimage-core` to `scrimage-scala` for the Scala-specific dependencies. This updates the library, and replaces the one usage of it using the latest syntax.

EDIT: The scope of this has expanded slightly to also fixing the build, by rewriting the `copyJARResourcesTo` function (now `copyResourcesFromFileSystem`) to work with arbitrary filesystems. While I'm not entirely sure what the cause of the build issues was, this issue might have had something to do with the specific JDK I am using (adoptopenjdk 8) as the issue was nonexistent before, and only popped up recently in the build.

### CODE REVIEW

**Review by BenFradet — DISMISSED** [review]
:+1:

**Comment by sloshy** [comment]
Currently working with the scripted tests, as these are failing now. The new copying resources code isn't quite matching up and I've got a lead on where to fix it at least. Hold off on reviewing for now.

**Comment by sloshy** [comment]
The build is "fixed" in terms of, if you pull this branch locally and run `sbt ci-test`, it will work assuming you have jekyll set up. Apparently there's a problem with the jekyll setup on GitHub Actions, so I've logged that in a comment on #531. For all intents and purposes I think this PR is technically reviewable again. Please run the tests locally if you want to confirm that it's just github having a jekyll problem.

**Review by rafaparadela — APPROVED** [review]

File: src/main/scala/microsites/util/MicrositeHelper.scala [hunk 1]
```diff
@@ -229,13 +254,26 @@ class MicrositeHelper(config: MicrositeSettings) {
...
-      .map { case (name, size) =>
-        (new File(s"$targetDir$jekyllDir/img/$name"), size)
-      }
-      .map { case (file, (width, height)) =>
-        Image.fromFile(sourceFile.toFile).scaleTo(width, height).output(file)
-      }
+    //This is a dirty classloader hack to allow the latest version of Scrimage to work.
+    //This plugin's default classloader is limited and cannot load ImageReader instances.
+    //We get a different ClassLoader that will work, replace it, and then set it back after we're done.
+    //Will be fixed in Scrimage 4.1.0, see: https://github.com/sksamuel/scrimage/issues/217
```
Comment (BenFradet) [inline src/main/scala/microsites/util/MicrositeHelper.scala:260]: maybe log an issue for when we upgrade to 4.1.0?

**Review by BenFradet — APPROVED** [review]

File: src/main/scala/microsites/util/MicrositeHelper.scala (hunk 1 above)
Comment (sloshy) [reply src/main/scala/microsites/util/MicrositeHelper.scala:260]: Logged as #535

**Pull request merged**